xhr/client/src/main/scala/endpoints4s/xhr/future/Endpoints.scala

```diff
@@ -17,7 +18,7 @@ trait Endpoints extends xhr.Endpoints with EndpointsWithCustomErrors
 trait EndpointsWithCustomErrors extends xhr.EndpointsWithCustomErrors {
 
   /** Maps `Result` to `Future` */
-  type Result[A] = Future[A]
+  case class Result[A](value: Future[A], abort: js.Function1[Unit, Unit])
```

Comment:
I was thinking that we should not try to make too many decisions about ergonomics, because users of the library might want to handle the calls in different ways, especially the streaming calls - a handler utility is basically a must for `ReadableStream` of `ChunkedEntities`.
Even for unary calls one might want to have something like:
```scala
// https://slinky.dev/docs/functional-components-and-hooks/
useEffect(
  () => {
    val abort = client
      .unaryCall(req)
      .handle(onSuccess, onFailure)
    () => abort()
  },
  Seq.empty
)
```
The least opinionated thing we can do is simply exposing the API. Does that make sense?

Reply:
Yes, but if I understood correctly here the only thing users want to do with a `Result` is to get its underlying `Future` value or maybe call `abort()` on it. I don’t think they will ever want to do `result.copy(abort = () => println())`?

Reply:
Oh yes, sorry for not clarifying - I totally agree with your suggested code bit, I default to case classes too often 😅 I was commenting more on the first part about the cancellable `Future` - seemed to me like the amount of work for benefit gained does not balance out well, or maybe I just do not have a good idea for it.

Reply:
No problem! Yes, I agree about the cancellable thing.